ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
+  echo "RELEASE_IMAGE_LATEST is an empty string, exiting"
+  exit 1
+fi
+
+export HOME=/tmp
+export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=${RELEASE_IMAGE_LATEST}
+# Ensure ignition assets are configured with the correct invoker to track CI jobs.
+export OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME_SAFE}/${BUILD_ID}
+
+echo "$(date -u --rfc-3339=seconds) - Creating reusable variable files..."
+# Create basedomain.txt
+echo "origin-ci-int-aws.dev.rhcloud.com" > "${SHARED_DIR}"/basedomain.txt
+base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+
+# Create clustername.txt
+echo "${NAMESPACE}-${JOB_NAME_HASH}" > "${SHARED_DIR}"/clustername.txt
+cluster_name=$(<"${SHARED_DIR}"/clustername.txt)
+
+# Create clusterdomain.txt
+echo "${cluster_name}.${base_domain}" > "${SHARED_DIR}"/clusterdomain.txt
+cluster_domain=$(<"${SHARED_DIR}"/clusterdomain.txt)
```

Comment:
why are these being provided as an API by this step.

Reply:
This is similar to what vSphere IPI does. I was trying to follow DRY.